pgx-tests/src/tests/array_tests.rs

```diff
@@ -99,6 +100,52 @@ fn return_zero_length_vec() -> Vec<i32> {
     Vec::new()
 }
 
+#[pg_extern]
+fn get_arr_nelems(arr: Array<i32>) -> libc::c_int {
+    // SAFETY: Eh it's fine, it's just a len check.
+    unsafe { RawArray::from_array(arr).unwrap().len() }
+}
+
+/// # Safety
+/// Don't call with an actually-null index, or you might see an invalid value.
+#[pg_extern]
+#[deny(unsafe_op_in_unsafe_fn)]
+unsafe fn get_arr_data_ptr_nth_elem(arr: Array<i32>, elem: i32) -> Option<i32> {
+    // SAFETY: this is Known to be an Array from ArrayType,
+    // and the index has to be a valid one.
+    unsafe {
```

Comment:
What is the reason for marking the fn as `unsafe` but also having the body of the function also wrapped in an `unsafe` block? The `get_arr_nelems` function above does not have it. Is it because this fn returns an `Option<>`?

Reply:
It's because I think we should assume that if a Postgres varlena array has "nullable indices", that we should treat the resulting slice as [`&[MaybeUninit<T>]`](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html) so it is incorrect to read the value of an index that was nulled.

The latter is because I want to move towards `#[deny(unsafe_op_in_unsafe_fn)]` across the repository.

Reply:
You should open an issue if you want to do this across the entire repo, @workingjubilee, so other people working with you can help with it.a

Reply:
Opened:
- https://github.com/tcdi/pgx/issues/653